### ISSUE TITLE
Error: The hook `orm` is taking too long to load.

### ISSUE THREAD

**Issue by insitusistemas** [dtoubelis/sails-cassandra#33]
**_This is the complete error:_**

info: Starting app...

error: Error: The hook `orm` is taking too long to load.
Make sure it is triggering its `initialize()` callback, or else set `sails.config.orm._hookTimeout to a higher value (currently 60000)
    at tooLong [as _onTimeout] (/usr/local/lib/node_modules/sails/lib/app/private/loadHooks.js:85:21)
    at Timer.listOnTimeout (timers.js:89:15)

/******************/

I am trying to connect to a cassandra remote server. It is accepting remote connections, I have connected succesfully using Datastax DevCenter.

Trying to solve this problem, I have made some changes to the models.js file to

module.exports.models = {
    connection: 'cassandra',
    migrate: 'safe'
};

and the developmen.js to 

module.exports = {
   models: {
     connection: 'cassandra'
   },
    hookTimeout: 60000
};

But the same error remains. Thanks a lot

**Comment by dleute** [comment]
Did you ever resolve this? I'm having the same problem. Does this adapter work with a newer cassandra?

**Comment by dtoubelis** [comment]
For some reasons I missed this issue, I will investigate it this weekend. Can you share what version of  node and waterline you are running?

**Comment by dleute** [comment]
node -v: v4.4.7
sails -v: 0.12.4
npm ls puts waterline@0.11.2

cassandra is 3.7.

At the very least the documentation setup sails-cassandra with "module" in connections instead of adapter. I wonder if there was a BC break?

Thanks!

**Comment by dtoubelis** [comment]
Yep, I can confirm that this is BC issue. If I run the test suite against Cassandra 3.x it fails with exactly same symptoms. Now, the question is how to fix it... Upgrading the driver does not help, so the only sensiple thing to do is to re-implement this module for Cassandra 3.x. I will try to do it this weekend and I will put notice in readme for this project.

**Comment by dtoubelis** [comment]
I'm closing it for now but feel free to comment.

**Comment by dleute** [comment]
Is Cassandra 3 that much different? I actually don't know the version of Cassandra that we will be using in production. If it can't work for the 2.x and 3.x versions will you put them in separate branches?

Also, if you need me to test anything, let me know.

**Comment by dtoubelis** [comment]
It is stated on [this page](http://docs.datastax.com/en/developer/driver-matrix/doc/nodejsDrivers.html#nodejs-drivers) that 2.x drivers are incompatible with 3.x database. However, even though 3.x drivers can work with 2.x database, Cassandra 3.x has few new data types. I'm going to implement them (long overdue)  but the adapter will likely become not be backward compatible with 2.x just because of that. I'm still investigating this but most likely I will fork the repo and create a separate `sails-cassandra3` package.

**Comment by dleute** [comment]
That makes sense. That also may mean 2.x could work for the time being. I will check what version of Cassandra the environment uses. 

I will try that on Monday.

**Comment by dtoubelis** [comment]
Okay I've pushed first cut of [sails-cassandra3](https://www.npmjs.com/package/sails-cassandra3) to npm. There is still some work to be done on it but it passes all unit tests, so give it a try.

**Comment by dleute** [comment]
It does in fact connect now, but I can't get updates or deletes to work. Do you need to use a different url syntax because of the UUID for an id?

It gives this error for updates: ```ResponseError: PRIMARY KEY part id found in SET part```

**Comment by dtoubelis** [comment]
Can I ask you to create a separate issue against [sails-cassandra3](https://github.com/dtoubelis/sails-cassandra3) and provide a code sample?

**Comment by dleute** [comment]
Sure. Here it is: dtoubelis/sails-cassandra3#4 

Let me know if you have more questions.

**Comment by dtoubelis** [comment]
Thanks. I'll look into it.

**Comment by dtoubelis** [comment]
BTW, I implemented support for Cassandra 3.x in the main adapter and removed `sails-cassandra3` from npm (and git). You may have to go back to this one. Sorry about that.

**Comment by dleute** [comment]
Not a problem. Thanks for the help!

**Comment by dleute** [comment]
Also, I got your suggestion on the cassandra orm. That is actually how I found sails/waterline (found your sails-cassandra repo and netflix staash). The thing is I'm not really looking for an ORM. I really just want to expose data tables via an API with the least amount of work. This requires understanding the structure of the data (like an orm does) but only to expose a generic api.

I'm not even really looking for the other things an orm does, like data migrations.

Anyway, thanks for all your help! I'll be giving this thing a test run this week.

--Derrek
